2018-05-04  Saibalaji Atmakuri  <saibalaj@lnxsrv06.seas.ucla.edu>

	* timespec.h (timespec_cmp): Fixed function to rely on comparisons
        rather than on integer subtractions to avoid integer overflow
	
--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2018-05-04 00:29:05.357118000 -0700
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec) { return -1; }
+  else if (b.tv_sec < a.tv_sec) { return 1; }
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
